libraries/APM_Control/AR_AttitudeControl.cpp

```diff
@@ -239,6 +246,37 @@ float AR_AttitudeControl::get_steering_out_rate(float desired_rate, bool skid_st
     return constrain_float(p + i + d, -1.0f, 1.0f);
 }
 
+// get latest desired turn rate in rad/sec (recorded during calls to get_steering_out_rate)
+float AR_AttitudeControl::get_desired_turn_rate() const
+{
+    // return zero if no recent calls to turn rate controller
+    if ((_steer_turn_last_ms == 0) || ((AP_HAL::millis() - _steer_turn_last_ms) > AR_ATTCONTROL_TIMEOUT)) {
+        return 0.0f;
+    }
+    return _desired_turn_rate;
+}
+
+// get latest desired lateral acceleration in m/s/s (recorded during calls to get_steering_out_lat_accel)
+float AR_AttitudeControl::get_desired_lat_accel() const
+{
+    // return zero if no recent calls to lateral acceleartion controller
+    if ((_steer_lat_accel_last_ms == 0) || ((AP_HAL::millis() - _steer_lat_accel_last_ms) > AR_ATTCONTROL_TIMEOUT)) {
+        return 0.0f;
```

Comment:
This is in-band.  Return a bool or a quiet_nan?

I'm actually a bit uneasy with these timeouts in a controller...

Reply:
these are for reporting only so the timeout should be fine.  what you'll see in the logging is the lateral acceleration will persist a touch longer in the logs than it actually should.  So for example, we could potentially see a mode change to MANUAL but still see a desired lateral accel in the logs for one message.

In practice I don't think it's a problem.  The alternative is to have the mode objects turn on/off controllers... and we will end up with small bugs where we forget to turn them off.